editor.py

```diff
@@ -66,25 +72,41 @@ def editor(initial_contents=None, filename=None, editor=None):
 
       editor
         The path to an editor to call.  If None, use editor.default_editor()
+        If None, use editor.default_editor().
+        If `editor` is a string, it will be split using `shlex`
+        If it is a list or tuple of strings, it will be used as is.
     """
     editor = editor or default_editor()
-    if not filename:
-        with tempfile.NamedTemporaryFile(mode='r+', suffix='.txt') as fp:
-            if initial_contents is not None:
-                fp.write(initial_contents)
-                fp.flush()
-
-            subprocess.call([editor, fp.name])
-
-            fp.seek(0)
-            return fp.read()
-
-    path = Path(filename)
-    if initial_contents is not None:
-        path.write_text(initial_contents)
-
-    subprocess.call([editor, filename])
-    return path.read_text()
+    if isinstance(editor, str):
+        editor = shlex.split(editor)
+    else:
+        try:
+            len(editor) and editor[0] or True and ''.join(editor)
```

Comment:
This is a test to make sure that `editor` is as we expect it.  The test is "correct" but now you point it out, I think it's sort of useless - we should just allow `subprocess.call`

Here's the reasoning - I'm making these claims:

* `editor` has a length (a tuple, a dict, a list)
* if the length is non-zero then it has an element at index `0`
* and each element is a string (because they can be joined together).

So the test is logically correct.  It's just not very useful.  :-D

I think I'm going to remove it!